### ISSUE TITLE
Local DNS IPv6 from public

### ISSUE THREAD

**Issue by Night1** [pi-hole/pi-hole#3433]
Hey

Seams to be a bug with Local DNS record, where  if the domain exist in a public dns already and has IPv6 either from CNAME or AAAA record, that is returned to pihole clinets. as IPv6 is preferred of IPv4  it might brake something for some people. 

PiHole 5.0 

; <<>> DiG 9.11.5-P1-1ubuntu2.6-Ubuntu <<>> apps.domain.gtld aaaa +noall +answer apps.niklas.pw +noall +answer
;; global options: +cmd
apps.domain.gtld.         147     IN      AAAA    2606:4700:3034::681b:8042
apps.domain.pw.         147     IN      AAAA    2606:4700:3037::681b:8142
apps.domain.gtld.         2       IN      A       192.168.1.243

**Comment by dschaper** [comment]
We'll need the debug token that was requested in the template in order to assist you. You can post a comment on https://discourse.pi-hole.net if you prefer not to.

**Comment by Night1** [comment]
https://tricorder.pi-hole.net/1kpxd4m7vu @dschaper

**Comment by Night1** [comment]
Any update on this?

**Comment by dschaper** [comment]
Sorry, missed this. 

It's working as intended and how DNS is designed. If you want the AAAA records to point to somewhere else then you'll need to configure that. Other options are to disable IPv6 or wait until the updates that will allow all AAAA records to be blocked is released.

**Comment by Night1** [comment]
Would that update block IPv6 only for those which are in local DNS or all IPv6 global AAAA. 
I can agree that this is not a bug. However it can cause confusion for those not familiar with IPv6.

Quick fix would be as you say is to add quad A but not always as desired since different webpage depending on origin  and local dns was used to test that.

**Comment by dschaper** [comment]
AAAA records would not resolve. 

There's really not much a DNS server can do, if a client asks for AAAA and it knows AAAA then it will respond AAAA. There is no relation between A and AAAA. You could set up unbound as the upstream and change the unbound configuration but you'll still need to configure an AAAA record to return, either public or private.

What address would you like returned for AAAA records for this situation?

**Comment by Night1** [comment]
NXDOMAIN would be fine, if local is setup with only single A or quad A since then it will signify that it is only on IPv4 or IPv6 asnd is within RFC for a respons when it is only one or  the other

**Comment by dschaper** [comment]
You could add a config line in a configuration file, like `/etc/dnsmasq.d/99-no-AAAA.conf` and have the contents as:

```plain
address=/example.com/::
```

That will return `::` for the domain.

**Comment by Night1** [comment]
hmm would that no return :: for any domain that does not have Quad A either direct or through CNAME?

**Comment by dschaper** [comment]
No, that would return `::` for the domain `example.com`. You want to return a A for a specific domain and no AAAA for that same domain, then use that domain in the `address` line.

**Comment by dschaper** [comment]
Any domain lacking AAAA, either direct or though CNAME wouldn't have an IP anyways.

**Comment by Night1** [comment]
makes sense, ill try it out. Thank you 💯 

would be nice to have an option for this in settings aswell for local records if pihole should forgo response of Quad A if a domain is in local with A only and other way around if only Quad A is there.

**Comment by dschaper** [comment]
That would be a feature request so the best place for that is at our [Discourse Forum](https://discourse.pi-hole.net/c/feature-requests/8).
